server/middleware/setUpWebSession.ts

```diff
@@ -1,19 +1,24 @@
-import redis from 'redis'
-import session from 'express-session'
 import connectRedis from 'connect-redis'
-import addRequestId from 'express-request-id'
 import express, { Router } from 'express'
+import addRequestId from 'express-request-id'
+import session from 'express-session'
+import { createClient } from 'redis'
 
 import config from '../config'
 
 const RedisStore = connectRedis(session)
 
-const client = redis.createClient({
-  port: config.redis.port,
+const redisURL =
+  config.redis.tls_enabled === 'true'
+    ? `rediss://${config.redis.host}:${config.redis.port}`
+    : `redis://${config.redis.host}:${config.redis.port}`
+
+const client = createClient({
+  url: redisURL,
   password: config.redis.password,
-  host: config.redis.host,
-  tls: config.redis.tls_enabled === 'true' ? {} : false,
+  legacyMode: true,
 })
+client.connect()
```

Comment:
A couple of thoughts here. This is very similar to the default client created by `new TokenStore()` by default when client is not passed. This has the additional `legacyMode: true` option.

Do you reckon it would be bad if we don't create the client explicitly here and rely on the default client to be instantiated there?

Also, given this Redis `client` constant is only used by the `TokenStore` instance and given that its `setToken()`/`getToken()` methods call `connect()` if the connection is not established, I wonder if this `client.connect()` here could also go.

Unless I'm missing something (which is entirely possible).

Reply:
this one is in legacy mode because connect-redis has not yet been updated for redis v4 so expects to use the callback form.

there's nowhere to put the `client.connect()` where the context is async. here it's top-level so await is not possible in CommonJS. the function that uses this client is non-async. connect-redis’s store is also non-async.

there's just nowhere to put this connect where it can be awaited :(

Reply:
Sorry, ignore me, I got confused between `RedisStore` and our `TokenStore` 🤦 

Both take a `RedisClient` but we don't have control over `RedisStore` and that wouldn't instantiate a new Redit client with the right configuration if not passed!